api/envoy/config/route/v3/route_components.proto

```diff
@@ -1351,6 +1351,24 @@ message RouteAction {
     // If the final output of the host rewrite is empty, then the update will be ignored and the
     // original host header will be preserved.
     string host_rewrite = 44;
+
+    // Indicates that during forwarding, the host header will be swapped with
```

Comment:
high-level API style comment: oneof is discouraged as it prevents servers from sending more than one type of config knob back to the client (that may or may not support this).
I think in this case, as there's already a oneof, it may make sense to allow that for now.

Reply:
Don't disagree with the API-style comment, but it's also stylish to align with existing style. I will keep this the way that it is.